core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -124,14 +124,19 @@ final case class Uri(
         renderScheme(s)
 
       case Uri(None, Some(a), _, _, _) =>
-        writer << a
+        writer << "//" << a // https://stackoverflow.com/questions/64513631/uri-and-double-slashes/64513776#64513776
 
       case Uri(None, None, _, _, _) =>
     }
 
     this match {
       case Uri(_, Some(_), p, _, _) if p.nonEmpty && !p.startsWith("/") =>
         writer << "/" << p
+      case Uri(None, None, p, _, _) =>
+        if (p.contains(":"))
+          writer << "./" << p // https://tools.ietf.org/html/rfc3986#section-4.2 last paragraph
```

Comment:
I've probably read that RFC a dozen times and never noticed this rule. Great find.